common/lib/xmodule/xmodule/modulestore/xml_importer.py

```diff
@@ -1167,5 +1168,5 @@ def _update_module_location(module, new_location):
     # explicitly set each field to its current value before triggering the save.
     if len(rekey_fields) > 0:
         for rekey_field_name in rekey_fields:
-            setattr(module, rekey_field_name, getattr(module, rekey_field_name))
+            module.fields[rekey_field_name].force_dirty(module)
```

Comment:
Can you clarify why using `setattr` wasn't doing the right thing. Using the setter on a field should result in the field being marked as dirty. If it doesn't do that, please provide a test-case (and ideally, a fix) in the XBlock repo.